gateway/modules/crud/dataloader.go

```
@@ -67,20 +76,33 @@ func (holder *resultsHolder) fillResults(res []interface{}) {
 		}
 
 		// Get the where clause
-		whereClause := holder.whereClauses[index]
-
+		meta := holder.metas[index]
+		isOperationTypeOne := meta.op == utils.One
+		var result interface{}
 		docs := make([]interface{}, 0)
 		for _, doc := range res {
-			if utils.Validate(whereClause.(map[string]interface{}), doc) {
+			if utils.Validate(meta.whereClause, doc) {
 				docs = append(docs, doc)
 			}
+			if isOperationTypeOne {
+				break
+			}
 		}
 
 		// Increment the where clause index
 		index++
 
+		if isOperationTypeOne {
+			if len(docs) >= 1 {
+				result = docs[0]
+			} else {
+				result = map[string]interface{}{}
```

Maybe you should throw an error here or make it nil